### PR TITLE
added kogito quickstart

### PR DESCRIPTION
**Pull request by mswiderski**
**Please make sure that your pull request is targeting the `development` branch.**

### CODE REVIEW

File: using-kogito/src/main/resources/META-INF/resources/index.html [hunk 1]
```diff
@@ -0,0 +1,152 @@
+<!DOCTYPE html>
```
Comment (cescoffier) [inline using-kogito/src/main/resources/META-INF/resources/index.html]: This file is not very useful.

File: using-kogito/src/main/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Configuration file
```
Comment (cescoffier) [inline using-kogito/src/main/resources/application.properties]: It the configuration is not used, maybe it's better to delete the file.

File: using-kogito/src/test/java/org/acme/kogito/PersonProcessTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import org.junit.jupiter.api.Test;
+import org.kie.kogito.Application;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.http.ContentType;
+
+@QuarkusTest
+public class PersonProcessTest {
+	
+	@Inject
```
Comment (cescoffier) [inline using-kogito/src/test/java/org/acme/kogito/PersonProcessTest.java]: Cosmetic - indentation.

**Comment by mswiderski** [comment]
@cescoffier thanks, updates done

**Comment by rsvoboda** [comment]
@mswiderski @cescoffier  I see test failures when using this quickstart

```
[ERROR] Failures:
[ERROR]   PersonProcessTest.testAdult:29 1 expectation failed.
Expected status code <200> but was <405>.

[ERROR]   PersonProcessTest.testChild:42 1 expectation failed.
Expected status code <200> but was <405>.

[INFO]
[ERROR] Tests run: 2, Failures: 2, Errors: 0, Skipped: 0
```

**Comment by mswiderski** [comment]
Let me have a look at it deeper tomorrow but it looks like some bug in kogito rather than QuickStart itself

**Comment by gsmet** [comment]
yes, it doesn't work, same here

**Comment by gsmet** [comment]
This is blocking the release so please give this some priority as I won't release with a quickstart not working.

Two issues:
- we have a ton of warnings such as:
```
2019-06-26 16:00:30,690 WARN  [org.dro.cor.xml.ExtensibleXmlParser] (build-10) Process Info: id:persons, pkg:org.acme.kogito, name:Person Process, version:null 
Node Info: id: name: 
Parser message: (null: 36, 33): schema_reference.4: Failed to read schema document 'BPMN20.xsd', because 1) could not find the document; 2) the document could not be read; 3) the root element of the document is not <xsd:schema>.
2019-06-26 16:00:30,691 WARN  [org.dro.cor.xml.ExtensibleXmlParser] (build-10) Process Info: id:persons, pkg:org.acme.kogito, name:Person Process, version:null 
Node Info: id:DataInputAssociation_1 name: 
Parser message: (null: 39, 63): schema_reference.4: Failed to read schema document 'BPMN20.xsd', because 1) could not find the document; 2) the document could not be read; 3) the root element of the document is not <xsd:schema>.
2019-06-26 16:00:30,692 WARN  [org.dro.cor.xml.ExtensibleXmlParser] (build-10) Process Info: id:persons, pkg:org.acme.kogito, name:Person Process, version:null 
Node Info: id: name: 
Parser message: (null: 40, 26): schema_reference.4: Failed to read schema document 'BPMN20.xsd', because 1) could not find the document; 2) the document could not be read; 3) the root element of the document is not <xsd:schema>.
```
This is not acceptable IMHO.
- and the other issue @rsvoboda mentioned is probably due to this NPE:
```
2019-06-26 16:00:36,522 ERROR [io.und.req.io] (executor-thread-1) Exception handling request 9a93ab6f-c068-46e1-a7e6-aee9563e3f38-2 to /persons: org.jboss.resteasy.spi.UnhandledException: org.jbpm.workflow.instance.WorkflowRuntimeException: [persons:2 - Evaluate person:2] -- null
	at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:106)
	at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:372)
	at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:209)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:496)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:252)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:153)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:362)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:156)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:238)
	at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:234)
	at io.quarkus.resteasy.runtime.ResteasyFilter.doFilter(ResteasyFilter.java:30)
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84)
	at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)
	at io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:68)
	at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
	at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:132)
	at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
	at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
	at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
	at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
	at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:292)
	at io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:81)
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:138)
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:135)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at io.quarkus.undertow.runtime.UndertowDeploymentTemplate$8$1$1.call(UndertowDeploymentTemplate.java:482)
	at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:272)
	at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:81)
	at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:104)
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:364)
	at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:830)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1538)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1429)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:32)
	at java.lang.Thread.run(Thread.java:748)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
Caused by: org.jbpm.workflow.instance.WorkflowRuntimeException: [persons:2 - Evaluate person:2] -- null
	at org.jbpm.workflow.instance.node.RuleSetNodeInstance.handleException(RuleSetNodeInstance.java:208)
	at org.jbpm.workflow.instance.node.RuleSetNodeInstance.internalTrigger(RuleSetNodeInstance.java:188)
	at org.jbpm.workflow.instance.impl.NodeInstanceImpl.trigger(NodeInstanceImpl.java:193)
	at org.jbpm.workflow.instance.impl.NodeInstanceImpl.triggerNodeInstance(NodeInstanceImpl.java:393)
	at org.jbpm.workflow.instance.impl.NodeInstanceImpl.triggerNodeInstance(NodeInstanceImpl.java:378)
	at org.jbpm.workflow.instance.impl.NodeInstanceImpl.triggerCompleted(NodeInstanceImpl.java:347)
	at org.jbpm.workflow.instance.node.StartNodeInstance.triggerCompleted(StartNodeInstance.java:72)
	at org.jbpm.workflow.instance.node.StartNodeInstance.internalTrigger(StartNodeInstance.java:43)
	at org.jbpm.workflow.instance.impl.NodeInstanceImpl.trigger(NodeInstanceImpl.java:193)
	at org.jbpm.ruleflow.instance.RuleFlowProcessInstance.internalStart(RuleFlowProcessInstance.java:38)
	at org.jbpm.process.instance.impl.ProcessInstanceImpl.start(ProcessInstanceImpl.java:233)
	at org.jbpm.workflow.instance.impl.WorkflowProcessInstanceImpl.start(WorkflowProcessInstanceImpl.java:456)
	at org.jbpm.process.instance.LightProcessRuntime.startProcessInstance(LightProcessRuntime.java:158)
	at org.jbpm.process.instance.LightProcessRuntime.startProcessInstance(LightProcessRuntime.java:167)
	at org.kie.kogito.process.impl.AbstractProcessInstance.start(AbstractProcessInstance.java:59)
	at org.acme.kogito.PersonsResource.createResource_persons(PersonsResource.java:40)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:151)
	at org.jboss.resteasy.core.MethodInjectorImpl.lambda$invoke$3(MethodInjectorImpl.java:122)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:602)
	at java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:614)
	at java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:1983)
	at java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:110)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:122)
	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:568)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:442)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:396)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:362)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:398)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:367)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invoke$1(ResourceMethodInvoker.java:341)
	at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:981)
	at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2124)
	at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:110)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:341)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:477)
	... 43 more
Caused by: java.lang.NullPointerException
	at org.acme.kogito.PersonsProcess.lambda$legacyProcess$0(PersonsProcess.java:57)
	at org.jbpm.workflow.instance.node.RuleSetNodeInstance.internalTrigger(RuleSetNodeInstance.java:105)
	... 80 more
```

**Comment by gsmet** [comment]
@mswiderski I'm postponing the release to tomorrow. I need a fix by tomorrow 11 AM Paris time or I'll revert this PR and we'll include it another release.

Sorry for rushing you but we really can't ship with a totally failing quickstart.

/cc @mariofusco

Thanks.

**Comment by mswiderski** [comment]
@gsmet totally understand it and working on it. It requires new release of kogito and by that update in bom of quarkus. So if that can't be done for this release please revert it and don't block the release

**Comment by gsmet** [comment]
@mswiderski it can be done if you can push Kogito to Central before that :).

**Comment by mswiderski** [comment]
@gsmet will do my best

**Comment by kmacedovarela** [comment]
Hi all, I also noticed the README in this PR is missing the Kogito project. I understand the hurry on this release process, but I don't know the best way to contribute to it atm. Here's a suggestion for the text. Feel free to improve it or change it in any way.

* [Using Kogito ](./using-kogito): Using Kogito: A business application with Kogito capable of natively executing business assets (i.e. business processes and drools rules).

Regards :)

**Comment by mswiderski** [comment]
@gsmet here is the PR for upgraded and fixed kogito to make it work https://github.com/quarkusio/quarkus/pull/2994 all available in maven central

**Comment by gsmet** [comment]
@kmacedovarela ah, thanks for mentioning it. I fixed it this morning, it will be pushed to the master branch in a few hours.

**Comment by gsmet** [comment]
BTW, the issues I mentioned earlier are all fixed thanks to the Kogito upgrade.

**Pull request merged**